backend/src/test/java/org/cryptomator/hub/license/LicenseHolderTest.java

```diff
@@ -157,6 +170,160 @@ public void testSetInvalidToken() {
 		}
 	}
 
+	@Nested
+	@TestProfile(LicenseHolderInitPropsTest.ValidInitPropsInstanceTestProfile.class)
+	@DisplayName("Testing LicenseHolder methods using InitProps")
+	class LicenseHolderInitPropsTest {
+
+		@InjectMock
+		Session session;
+
+		@InjectMock
+		LicenseValidator validator;
+
+		MockedStatic<Settings> settingsClass;
+
+		public static class ValidInitPropsInstanceTestProfile implements QuarkusTestProfile {
+			@Override
+			public Map<String, String> getConfigOverrides() {
+				return Map.of("hub.initial-id", "42", "hub.initial-license", "token");
+			}
+		}
+
+		@BeforeEach
+		public void setup() {
+			Query mockQuery = Mockito.mock(Query.class);
+			Mockito.doNothing().when(session).persist(Mockito.any());
+			Mockito.when(session.createQuery(Mockito.anyString())).thenReturn(mockQuery);
+			Mockito.when(mockQuery.getSingleResult()).thenReturn(0l);
+
+			settingsClass = Mockito.mockStatic(Settings.class);
+		}
+
+		@AfterEach
+		public void teardown() {
+			settingsClass.close();
+			Arc.container().instance(LicenseHolder.class).destroy();
+		}
+
+		@Test
+		@DisplayName("If init token is valid, set it in license holder")
+		public void testValidInitTokenSet() {
+			var decodedJWT = Mockito.mock(DecodedJWT.class);
+			Mockito.when(validator.validate("token", "42")).thenReturn(decodedJWT);
+			Settings settingsMock = new Settings();
+			settingsMock.hubId = "42";
+			settingsClass.when(Settings::get).thenReturn(settingsMock);
+
+			holder.init();
+
+			// init implicitly called due to @PostConstruct which increases the times to verify by 1
+			// See https://github.com/cryptomator/hub/pull/229#discussion_r1374694626 for further information
+			Mockito.verify(validator, Mockito.times(2)).validate("token", "42");
+			Assertions.assertEquals(decodedJWT, holder.get());
+		}
+
+		@Test
+		@DisplayName("If init token is invalid, do net set it in license holder and nullify db entry")
```

Comment:
The description is misleading, the text can be interpret as "if token is invalid, set db entry to null, no matter what there was before".

What you actually want to say is: "If init token is invalid and no db entry exists, do not modify db" (or something like that).